src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -0,0 +1,112 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.gson.Gson;
+import com.google.sps.data.Event;
+import com.google.sps.Trip;
+import com.google.sps.TripDay;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that gets events from datastore and writes them to json
+ * used from frontend to create events
+ */
+@WebServlet("/get-calendar")
+public class CalendarServlet extends HttpServlet {
+
+  /**
+   * Adam TODO: put functionality into a Utility class!
+   */
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) 
+      throws IOException {
+    response.setContentType("application/json;");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String stringTripKey = request.getParameter("tripKey");
+    Key tripKey = KeyFactory.stringToKey(stringTripKey);
+
+    // get current user
+    Entity userEntity = AuthServlet.getCurrentUserEntity();
+
+    // gets the events from datastore and writes them to .../get-calendar
+    doGetEvents(response, datastore, userEntity, tripKey);
+  }
+
+  /**
+   * Gets the events and prints them to writer, necessary break up for testing
+   */
+  public void doGetEvents(HttpServletResponse response, 
+      DatastoreService datastore, Entity userEntity, Key tripEntityKey) throws IOException {
+  
+    Filter tripKeyFilter =
+      new FilterPredicate("__key__", FilterOperator.EQUAL, tripEntityKey);
+    Query tripQuery = new Query(Trip.TRIP, userEntity.getKey());
+    tripQuery.setFilter(tripKeyFilter);
+
+    PreparedQuery tripResults = datastore.prepare(tripQuery);
+    List<Entity> listResults = tripResults.asList(FetchOptions.Builder.withDefaults());
+
+     if (listResults.isEmpty()) {
+      throw new IllegalArgumentException("Trip does not exist");
+    }
+
+    // get the Entity object of the trip.
+    Entity tripEntity =  listResults.get(0);
```

Comment:
If you are only expecting one result here, you should probably use the asSingleEntity method in PreparedQuery: https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#assingleentity